README.md

```diff
@@ -0,0 +1,754 @@
+| Collection negative index | `[-1]`, `[-42]`            | The element by index from the end. `-1` is the index for the last element.                      |
+| Try                       | `.foo?`, `["nope"]?` | Returns `null` on what would otherwise fail                                                     |
+
+Any selection MAY begin and/or end with a single dot. Multiple dots (e.g. `..`, `...`) MUST NOT be used anywhere in a selector.
+
+The try operator is idempotent, and repeated tries (`.foo???`) MUST be treated as a single one.
+
+For example, consider the following `args` from an `Invocation`:
+
+``` json
+{
+  "args": {
+    "from": "alice@example.com",
+    "to": ["bob@example.com", "carol@not.example.com", "dan@example.com"],
+    "cc": ["fraud@example.com"],
+    "title": "Meeting Confirmation",
+    "body": "I'll see you on Tuesday"
+  }
+}
+```
+
+<table>
+<!-- 
+NOTE: You cannot add _any_ indentation to this table if you want
+      GitHub to render that big multiline code block correctly 
+-->
+<thead>
+<tr>
+<td>Selector</td>
+<td>Returned Value</td>
+</tr>
+</thead>
+  
+<tbody>
+<tr>
+<td><pre>"."</pre></td>
+<td>
+
+```json
+{
+  "from": "alice@example.com",
+  "to": ["bob@example.com", "carol@not.example.com", "dan@example.com"],
+  "cc": ["fraud@example.com"],
+  "title": "Meeting Confirmation",
+  "body": "I'll see you on Tuesday"
+}
+```
+  
+</td>
+</tr>
+  
+<tr>
+<td><pre>".title"</pre></td>
+  
+<td>
+
+``` json
+"Meeting Confirmation"
+```
+
+</td>
+</tr>
+
+<tr>
+<td><pre>".cc"</pre></td>
+<td>
+
+``` json
+["fraud@example.com"]
+```
+
+</td>
+</tr>
+    
+<tr>
+<td><pre>".to[1]"</pre></td>
+<td>
+
+``` json
+"carol@not.example.com"
+```
+
+</td>
+</tr>
+
+<tr>
+<td><pre>".to[-1]"</pre></td>
+<td>
+
+``` json
+"dan@example.com"
+```
+
+</td>
+</tr>
+
+<tr>
+<td><pre>".to[99]?"</pre></td>
+<td>
+
+``` json
+null
+```
+
+</td>
+</tr>
+</tbody>
+</table>
+
+### Differences from jq
+
+[jq] is a much larger language than UCAN's selectors. jq includes features like pipes, arithmatic, regexes, assignment, recursive descent, and so on which MUST NOT be supported in the UCAN Policy language.
+
+jq produces streams of values, in contract to UCAN argument selectors which return an IPLD value. This introduces the primary difference between jq and UCAN argument selectors is how to treat output of the try (`?`) operator: UCAN's `try` selector operator MUST return `null` for the failure case.
+
+## Validation
+
+Validation involves substituting the values from the `args` field into the Policy, and evaluating the predicate. Since Policies are tree structured, selector substitution and predicate evaluation MAY proceed in any order.
+
+If a selector cannot be resolved (there is no value at that path), the associated statement MUST return false, and MUST NOT throw an exception. Note that for consistent semantics, selecting a missing keys on a map MUST return `null` (but nested selectors without a try MUST then fail the predicate).
```

Comment:
This contradicts some of the prior agreements that made here https://github.com/ucan-wg/delegation/issues/5 specifically I made case that:

1. reference to missing key
2. out of bound index reference
3. Iteration over non collection

All must be errors which could be trapped using try `?` suffix to opt-in into return `null` behavior. My rational had been that optionality could be handled explicitly.

Reply:
Yeah I think is a case of me working from the Rust version where we had diverged while iterating quickly. You're right and the rationale makes sense to me; will fix 👍

Reply:
Ah, it's also how `jq` behaves:

```shell
> echo '{"a": 1}' | jq '.b.c.d.e.f'
null

> echo '[1]' | jq '.b'
jq: error (at <stdin>:1): Cannot index array with string "b"
```

When writing this, I had `jq` running on the side to check edge cases. Anywhere that we diverge from `jq` will need to be highlighted in big neon lights (or we need to do whatever they do since it's easier to manually test against / principle of least surprise)